mm/2s2h/Enhancements/Trackers/ItemTracker/ItemTracker.cpp

```diff
@@ -0,0 +1,469 @@
+    }
+    ImVec2 textSize = ImGui::CalcTextSize(itemCount.c_str());
+
+    ImVec2 textPos =
+        ImVec2(currentPos.x + textureSize.x - textSize.x - 2.0f, currentPos.y + textureSize.y - textSize.y - 2.0f);
+    ImGui::SetCursorPos(textPos);
+    ImGui::SetWindowFontScale(scale);
+    ImGui::Text(itemCount.c_str());
+}
+
+void DrawItemSlot(int16_t itemId, float scale, bool isRandoItem) {
+    ImVec2 currentPos = ImGui::GetCursorPos();
+    TrackerImageObject imageObject = GetTextureObject(itemId, isRandoItem);
+    ImGui::Image(imageObject.textureId,
+                 ImVec2(imageObject.textureDimensions.x * scale, imageObject.textureDimensions.y * scale), ImVec2(0, 0),
+                 ImVec2(1, 1), imageObject.textureColor, ImVec4(0, 0, 0, 0));
+    UIWidgets::Tooltip(GetItemTrackerItemName(itemId, isRandoItem).c_str());
+    if (CVarGetInteger("gSettings.ItemTracker.ItemCounts", 0)) {
+        DrawItemCounts(itemId, isRandoItem, imageObject.textureDimensions * scale, scale, currentPos);
+    }
+}
+
+void DrawItemWindowList(TrackerItemListObject windowObject, bool isRandoItem) {
+    int columns = windowObject.columnLength;
+    if (windowObject.itemList.size() < windowObject.columnLength) {
+        columns = windowObject.itemList.size();
+    }
+
+    if (ImGui::BeginTable(windowObject.windowName.c_str(), columns)) {
+        ImGui::PushStyleVar(ImGuiStyleVar_ItemSpacing, ImVec2(5, 5));
+
+        for (auto& item : windowObject.itemList) {
+            ImGui::TableNextColumn();
+            ImVec2 framePadding = ImVec2(item >= ITEM_SONG_SONATA && item <= ITEM_SONG_SUN ? 8.0f : 0, 0);
+            ImGui::PushStyleVar(ImGuiStyleVar_FramePadding, framePadding);
+            DrawItemSlot(item, windowObject.windowScale, isRandoItem);
+            ImGui::PopStyleVar(1);
+        }
+
+        ImGui::PopStyleVar(1);
+        ImGui::EndTable();
+    }
+}
+
+void ItemTrackerWindow::Draw() {
+    if (!IsVisible()) {
+        return;
+    }
+
+    if (!gPlayState) {
+        return;
+    }
+
+    ImGuiWindowFlags windowFlags = ImGuiWindowFlags_AlwaysAutoResize | ImGuiWindowFlags_NoNav |
+                                   ImGuiWindowFlags_NoFocusOnAppearing | ImGuiWindowFlags_NoResize |
+                                   ImGuiWindowFlags_NoScrollWithMouse | ImGuiWindowFlags_NoScrollbar;
+
+    if (!CVarGetInteger("gSettings.ItemTracker.WindowType", 0)) {
+        windowFlags |= ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoDocking;
+    }
+    shouldWindowSplit = CVarGetInteger("gSettings.ItemTracker.WindowGroup", 0);
+
+    ImGui::PushStyleColor(ImGuiCol_Border, ImVec4(0, 0, 0, 0));
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 4.0f);
+
+    std::vector<std::vector<TrackerItemListObject>*> windows = {
+        &BenGui::mItemTrackerWindow->namedItemWindows,
+        &BenGui::mItemTrackerWindow->randoItemWindows,
+    };
+
+    uint32_t windowIndex = TRACKER_MAIN;
+    for (auto* window : windows) {
+        if ((*window).empty()) {
```

Comment:
If I'm understanding right, `auto* window` clarifies that window is a pointer of some type, in which case, this might be cleaner:

```suggestion
        if (window->empty()) {
```